### PR TITLE
cli: display multiwallet total balance in -getinfo

### PR DESCRIPTION
**Pull request by jonatack**
Per review suggestion in https://github.com/bitcoin/bitcoin/pull/18594#issuecomment-632026269, this PR adds a `total_balance` field for `-getinfo` in multiwallet mode.

```json
  "balances": {
    "": 0.00001000,
    "Encrypted": 0.00003500,
    "day-to-day": 0.00000120,
    "side project": 0.00000094
  },
  "total_balance": 0.00003714
}
```

Updated help:
```
$ bitcoin-cli -h | grep -A3 getinfo
  -getinfo
       Get general information from the remote server, including the total
       balance and the balances of each loaded wallet when in
       multiwallet mode. Note that -getinfo is the combined result of
       several RPCs (getnetworkinfo, getblockchaininfo, getwalletinfo,
       getbalances, and in multiwallet mode, listwallets), each with
       potentially different state.
```

Credit to João Barbosa for the suggestion.

### CODE REVIEW

**Comment by promag** [comment]
Concept ACK.

Could update doc?

**Comment by jonatack** [comment]
> Concept ACK.
> 
> Could update doc?

Thanks @promag. Added a release note update.

**Comment by promag** [comment]
This it `--help` output
```
  -getinfo
       Get general information from the remote server. Note that unlike
       server-side RPC calls, the results of -getinfo is the result of
       multiple non-atomic requests. Some entries in the result may
       represent results from different states (e.g. wallet balance may
       be as of a different block from the chain state reported)
```
nit, I wonder it this should document the result.

**Comment by jonatack** [comment]
Ok, I wondered about that. I'll push an update tomorrow to it; something very short about wallet balances and also fix the odd "be as of a".

**Comment by Wiphawee112** [comment]
ฉันมียอดที่ได้รับบริจาคหรือไม่ ฉันจะนำมันออกมาได้อย่างไร

**Comment by jonatack** [comment]
> I wonder it this should document the result.

Made an attempt at improving it with 57f771b.

**Comment by vahidjalaliii** [comment]
#19089

**Comment by luke-jr** [comment]
"balance" seems to perhaps oversimplify things...

Maybe "total_balance"?

**Comment by Wiphawee112** [comment]
ฉันจะทราบได้อย่างไร่่่่ะว่ายอดเงินอยุ่ทึ่ใหน ช่วยแนะนำด้วยค่ะ

**Comment by jonatack** [comment]
> ฉันจะทราบได้อย่างไร่่่่ะว่ายอดเงินอยุ่ทึ่ใหน ช่วยแนะนำด้วยค่ะ

"I can know exactly what the balance is. Please help."

I'll take that as a Concept ACK :)

**Comment by jonatack** [comment]
> "balance" seems to perhaps oversimplify things...
> 
> Maybe "total_balance"?

That was my initial thought as well. @promag WDYT?

**Comment by promag** [comment]
Also looks fine to me. I just mentioned `balance` to keep it backward compatible, but I guess that's not a concern in -cli.

**Comment by jonatack** [comment]
Thanks @luke-jr and @promag, updated to `total_balance` displayed after the balances.

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -453,9 +454,29 @@ static UniValue ConnectAndCallRPC(BaseRequestHandler* rh, const std::string& str
...
+}
+
+static UniValue ValueFromAmount(const CAmount& amount)
+{
+    bool sign{amount < 0};
+    int64_t n_abs{sign ? -amount : amount};
+    int64_t quotient{n_abs / COIN};
+    int64_t remainder{n_abs % COIN};
+    return UniValue(UniValue::VNUM, strprintf("%s%d.%08d", sign ? "-" : "", quotient, remainder));
+}
```
Comment (luke-jr) [inline src/bitcoin-cli.cpp:474]: Can we get these into the `libbitcoin_util.a` module? I think this is the third copy...

Comment (jonatack) [reply src/bitcoin-cli.cpp:474]: Yes, third one. Agreed, a good follow-up.

Comment (promag) [reply src/bitcoin-cli.cpp:474]: Why not in this PR or in a base PR especially considering there aren't ACKs.

Comment (jonatack) [reply src/bitcoin-cli.cpp:474]: I think that refactoring would best be done in a separate PR.

**Comment by jonatack** [comment]
Good to see this is in Bitcoin Knots. Rebased, will close in 10 days if no ACKs.

**Comment by jonatack** [comment]
Closing, there are many open PRs and this has no ACKs. At least this PR was pulled into other Bitcoin implementations, so it will be useful for those users.

**Pull request closed without merging**